src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -68,6 +72,8 @@
     public static final String DESIRED_STATUS_CANCELED = "CANCELED";
     public static final String DEPLOYMENT_SHADOW_NAME = "AWSManagedGreengrassV2Deployment";
     @Inject
+    private Kernel kernel;
+    @Inject
```

Comment:
why is this added? How was it working before?

Reply:
this is just to get `DeploymentService.getCurrentDeploymentTaskMetadata`. We didn't need that info. But in order to tell why last deployment didn't finish, (unexpected failure vs. managed restart) we can check the ongoing deployment ID. If it was a managed restart, the ongoing deployment should be the same.